app/models/concerns/attachable.rb

```diff
@@ -0,0 +1,9 @@
+module Attachable
+  include Hashable
+
+  def encoded_url(variant)
```

Comment:
rename to permanent_attachment_url since that properly identifies the functionality